src/Microsoft.AspNetCore.OData/Results/CreatedODataResult.cs

```diff
@@ -23,22 +23,23 @@ namespace Microsoft.AspNet.OData.Results
     /// <remarks>This action result handles content negotiation and the HTTP prefer header. It generates a location
     /// header containing the edit link of the created entity and, if response has status code: NoContent, also
     /// generates an OData-EntityId header.</remarks>
-    public class CreatedODataResult<T> : IActionResult
+    public class CreatedODataResult<T> : ObjectResult
     {
         private readonly T _innerResult;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="CreatedODataResult{T}"/> class.
         /// </summary>
         /// <param name="entity">The created entity.</param>
-        public CreatedODataResult(T entity)
+        public CreatedODataResult(object entity)
+            :base(entity)
         {
             if (entity == null)
             {
                 throw Error.ArgumentNull("entity");
             }
 
-            this._innerResult = entity;
+            this._innerResult = (T)entity;
```

Comment:
https://github.com/OData/WebApi/blob/cc0786b6e445d917569e5cec0b294616b9d7f669/src/Microsoft.AspNetCore.OData/Results/CreatedODataResult.cs#L81-L84 also uses the ObjectResult.

We should use that to process.

Reply:
Maybe use `this._innerResult = entity as T;` and then handle the scenario where the outcome is `null`. If it's acceptable for `entity` to be null, `entity as T` would yield null but that wouldn't be because an error was experienced when attempting to cast `entity` into a `T`.

Reply:
I reverted this change